File: solutions/go/rna-transcription/2/rna_transcription.go
package strand

func ToRNA(dna string) (rna string) {
	m := map[rune]string{
		'G': "C",
		'C': "G",
		'T': "A",
		'A': "U",
	}
	for _, d := range dna {
		rna += m[d]
	}
	return
}

File: solutions/zig/rna-transcription/2/rna_transcription.zig
const mem = @import("std").mem;

pub fn toRna(allocator: mem.Allocator, dna: []const u8) mem.Allocator.Error![]const u8 {
    const rna = try allocator.alloc(u8, dna.len);
    for (dna, rna) |n, *r| {
        r.* = switch (n) {
            'A' => 'U',
            'C' => 'G',
            'G' => 'C',
            'T' => 'A',
            else => unreachable,
        };
    }
    return rna;
}
